internal/kallsyms/kallsyms.go

```diff
@@ -72,3 +74,56 @@ func loadKernelModuleMapping(f io.Reader) (map[string]string, error) {
 	}
 	return mods, nil
 }
+
+func LoadSymbolAddresses(symbols []string) (map[string]uint64, error) {
```

Comment:
I don't think this function should return all symbols, as currently we'll happily take an address of any kernel symbol (like a function) and stuff it into an instruction. I hope the verifier rejects that, but we should filter kallsyms output to catch this earlier.

I can't find a reference for the kallsyms format, but this seems to be a variable: `00000000000000E0 V hv_root_partition`. Perhaps `LoadSymbolAddresses` and `loadSymbolAddresses` should take a string to match the type against and filter the output if the string is not `""`.

Reply:
I have to say, here I mostly went with what libbpf is doing ([source in kernel 5.15](https://github.com/torvalds/linux/blob/v5.15/tools/lib/bpf/libbpf.c#L6586)). 

Unless I overlooked something, they also do not check the `sym_type` at all. 

Is there any problem you think will happen if loading of any symbol is allowed?